driver-legacy/src/main/com/mongodb/LegacyMixedBulkWriteOperation.java

```diff
@@ -204,26 +148,24 @@ private WriteConcernResult translateBulkWriteResult(final BulkWriteResult bulkWr
     }
 
     private int getCount(final BulkWriteResult bulkWriteResult) {
-
-
         int count = 0;
-        if (getType() == UPDATE || getType() == REPLACE) {
+        if (type == UPDATE || type == REPLACE) {
             count = bulkWriteResult.getMatchedCount() + bulkWriteResult.getUpserts().size();
-        } else if (getType() == DELETE) {
+        } else if (type == DELETE) {
             count = bulkWriteResult.getDeletedCount();
         }
         return count;
     }
 
     private boolean getUpdatedExisting(final BulkWriteResult bulkWriteResult) {
-        if (getType() == UPDATE) {
+        if (type == UPDATE || type == REPLACE) {
```

Comment:
Tests uncovered this.  UpdateOperation claimed all updates where of type UPDATE, but in reality some are of type REPLACE.  Now that we're getting the type from the first write request, some updates are of type REPLACE and this condition was being incorrectly skipped.